src/components/CommonButtonInput/index.tsx

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { ReactNode } from 'react';
+import { UseFormRegister, FieldError, RegisterOptions, FieldValues, Path } from 'react-hook-form';
+
+import { TriangleAlertIcon } from 'lucide-react';
+
+import colors from '@/styles/color';
+
+import * as S from './CommonButtonInput.css';
+
+interface CommonButtonInputProps<T extends FieldValues> {
+  id: Path<T>; // react-hook-form의 Path 타입을 사용
+  label: string;
+  onClick?: () => void;
+  button: ReactNode;
+  buttonValue?: string;
+  type?: 'text' | 'number' | 'email' | 'password' | 'datetime-local';
+  placeholder?: string;
+  register?: UseFormRegister<T>; // 제네릭 타입으로 register 타입 설정
+  error?: FieldError;
+  validation?: RegisterOptions<T>; // 유효성 검사 옵션을 제네릭으로 설정
+  min?: number | string; // 데이트 피커에 활용
+  value?: string | number;
+  disabled?: boolean;
+}
+
+function CommonButtonInput<T extends FieldValues>({
+  id,
+  label,
+  type = 'text',
+  placeholder = '',
+  register,
+  error,
+  validation = {},
+  button,
+  min,
+  value,
+  disabled,
+}: CommonButtonInputProps<T>) {
+  console.log('CommonInput render', value);
+
+  return (
+    <label htmlFor={String(id)} className={S.label}>
+      <div className={disabled ? S.disabledTitle : S.title}>{label}</div>
+      <div className={S.inputWrapper}>
+        <input
+          disabled={disabled}
+          value={value}
+          className={S.input}
+          id={String(id)}
+          type={type}
+          min={min}
+          placeholder={placeholder}
+          {...(register ? register(id, validation) : {})}
+        />
+        {button}
+      </div>
+      {error && (
+        <span className={S.error}>
+          <TriangleAlertIcon size={16} stroke={colors.primary10} />
+          {error.message}
+        </span>
+      )}
+    </label>
```

Comment:
```suggestion
        <CommonInput
          id={id}
          label={label}
          type={type}
          placeholder={placeholder}
          register={register}
          error={error}
          validation={validation}
          min={min}
          value={value}
          disabled={disabled}
         >
           {button}
         </CommonInput>
```
뭔가 잘하면 이렇게도 할 수 있을 것 같아요

Reply:
저렇게 구조가 될려면 input에 button이라는 것을 받아야하는 구조가 되어야 합니다. 그렇다면 `CommonButtonInput`도 필요가 없어지겠지요. 합치게 된다면 wrapper와 같은 부분도 button의 유무에 따라서 조건부로 렌더링이 필요할 것 같습니다...
현재로써는 나누는 것이 오히려 깔끔하다고 판단이 되어서 위처럼 진행하겠습니다! 추후 리펙토링 및 코드 재정비에서 좋은 방법 있으면 같이 논의해보면 좋을 부분인 것 같습니다.